core/src/cgroup.rs

```diff
@@ -432,4 +436,10 @@ mod tests {
             .unwrap()
             .join(super::current_cgroup().unwrap())
     }
+
+    /// Generates a name for the current cgroup
+    fn gen_name() -> String {
+        let val: u64 = rand::random();
```

Comment:
Please use something that is reliably unique instead (e.g., a UUID). Or just the thread ID (`gettid`) of the test.

Reply:
+1 for uuid. https://docs.rs/uuid/latest/uuid/ looks good at a first glance

Reply:
I personally think that the UUID crate is overkill for a test dependency. 64-bit integers are totally sufficient in that case. I've added a loop on top of it, in order to avoid collisions, although the probability is with $2^{-64}$ very very low.

Reply:
Beside this, the UUID crate seems to require an instance before you can generate UUIDs. This would already make the code less obvious in my opinion, compared to the anonymous integer generator.